### PR TITLE
[ECO-2103] Add inputs for min out amounts

### PR DESCRIPTION
**Pull request by CRBl69**
# Description

This PR adds an advanced mode for swaps and liquidity provisioning.

The advanced mode adds a new field titled "Minimum output amount".

The user can then set a minimum output amount.

Whenever an other input changes, the minimum output amount is automatically updated.

image

image

image

image

image

image

image

image

image

# Testing

See vecrel.

# Checklist

- [x] Did you check all checkboxes from the linked Linear task?

### CODE REVIEW

**Comment by CRBl69** [comment]
I updated it to use percentages.

Now, whenever a user chooses to use percentages, this is what it looks like:

image

When clicking on the percent sign, it switches to this:

image

Note that the newly displayed value upon mode change is calculated as `prevPercentage * outputAmount` or `prevMinOut / outputAmount * 100`.

Also, percentage mode is enabled by default

File: src/typescript/frontend/src/lib/server-env.ts [hunk 1]
```diff
@@ -42,8 +42,8 @@ export const REVALIDATION_TIME: number = Number(process.env.REVALIDATION_TIME);
...
-if (APTOS_NETWORK !== Network.LOCAL && EMOJICOIN_INDEXER_URL.includes("localhost")) {
-  throw new Error(
-    `APTOS_NETWORK is ${APTOS_NETWORK} but the indexer processor url is ${EMOJICOIN_INDEXER_URL}`
-  );
-}
+//if (APTOS_NETWORK !== Network.LOCAL && EMOJICOIN_INDEXER_URL.includes("localhost")) {
+//  throw new Error(
+//    `APTOS_NETWORK is ${APTOS_NETWORK} but the indexer processor url is ${EMOJICOIN_INDEXER_URL}`
+//  );
+//}
```
Comment (xbtmatt) [inline src/typescript/frontend/src/lib/server-env.ts:49]: Can undo this and rebase

**Review by xbtmatt — COMMENTED** [review]
The design for the swap page is awesome! Personally I think it fits really well, maybe we can do it for the liquidity page, too?

I'm also open to the toggle on both as well, but I think it needs a label, not just a hover tooltip.

But let's go with the `+` design on both first. It's really intuitive.

Also I really like that you made a % input, was that in the figma or did you just add it? Looks great!

**Comment by CRBl69** [comment]
There is no Figma for this. Just freestyled.

**Review by xbtmatt — COMMENTED** [review]
Closing this PR in favor of waiting for Kirsten to make a new design for this.

Just want to note that I thought it was nice, and liked the initiative you took freestyling it :)

**Pull request closed without merging**